concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -1135,83 +1160,62 @@ handleUpdateBakerKeys wtc bkuElectionKey bkuSignKey bkuAggregationKey bkuProofSi
-                    -- account does not yet exist, so create it, but we need to be careful
-                    when (null keys || length keys > 255) $ throwError $ Just AccountCredentialInvalid
-                    let aaddr = ID.addressFromRegId regId
-                    case ID.values cdi of
-                      Nothing -> throwError $ Just AccountCredentialInvalid
-                      Just cdv -> do
-                        -- this check is extremely unlikely to fail (it would amount to a hash collision since
-                        -- we checked regIdEx above already).
-                        accExistsAlready <- lift (addressWouldClash aaddr)
-                        let check = AH.verifyCredential cryptoParams ipInfo arsInfos cdiBytes (Left messageExpiry)
-                        unless (not accExistsAlready && check) $ throwError $ Just AccountCredentialInvalid
-                        -- Add the account to the state, but only if the credential was valid and the account does not exist
-                        _ <- lift (createAccount cryptoParams aaddr cdv)
-                        mkSummary (TxSuccess [AccountCreated aaddr, CredentialDeployed{ecdRegId=regId,ecdAccount=aaddr}])
-  case res of
-    Left err -> return (TxInvalid <$> err)
-    Right ts -> return (Just ts)
-
+handleDeployCredential (WithMetadata{wmdData=cred@AccountCreation{messageExpiry=messageExpiry, credential=cdi}}, verRes) cdiHash = do
+    res <- runExceptT $ do
+      liftedCm <- lift getChainMetadata
+      let ts = slotTime liftedCm
+      -- check that the transaction is not expired
+      when (transactionExpired messageExpiry ts) $ throwError (Just ExpiredTransaction)
+      remainingEnergy <- lift getRemainingEnergy
+      let cost = Cost.deployCredential (ID.credentialType cdi) (ID.credNumKeys . ID.credPubKeys $ cdi)
+      when (remainingEnergy < cost) $ throwError Nothing
+      let mkSummary tsResult = do
+            tsIndex <- lift bumpTransactionIndex
+            return $ TxValid $ TransactionSummary{
+              tsSender = Nothing,
+              tsHash = cdiHash,
+              tsCost = 0,
+              tsEnergyCost = cost,
+              tsType = TSTCredentialDeploymentTransaction (ID.credentialType cdi),
+              ..
+              }
+
+      let regId = ID.credId cdi
+      let aaddr = ID.addressFromRegId regId
+      -- We always need to make sure that the account was not created in between
+      -- the transaction was received and the actual execution.
+      -- Check that the registration id does not exist
+      regIdEx <- lift (TVer.registrationIdExists regId)
+      when regIdEx $ throwError $ Just (DuplicateAccountRegistrationID (ID.credId cdi))
+      -- Check that the address would not clash with an existing one.
+      accExistsAlready <- lift (addressWouldClash aaddr)
+      when accExistsAlready $ throwError $ Just AccountCredentialInvalid
+      liftedCryptoParams <- lift TVer.getCryptographicParameters
```

Comment:
I'm not sure why you are naming these "lifted*". The type here is just cryptographic parameters. lift is just there since we are inside the ExceptT context.